src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlConverter.cs

```diff
@@ -1082,45 +1086,63 @@ public static int ToChars(DateTime value, byte[] chars, int offset)
             return offset - offsetMin;
         }
 
-        public static bool IsWhitespace(string s)
+        public static bool IsWhitespace(ReadOnlySpan<char> chars) =>
+            chars.IndexOfAnyExcept(s_whitespaceChars) < 0;
+
+        public static bool IsWhitespace(ReadOnlySpan<byte> bytes) =>
+            bytes.IndexOfAnyExcept(s_whitespaceBytes) < 0;
+
+        public static bool IsWhitespace(char ch) =>
+            ch is <= ' ' and (' ' or '\t' or '\r' or '\n');
+
+        public static int StripWhitespace(Span<char> chars)
         {
-            for (int i = 0; i < s.Length; i++)
+            int count = chars.IndexOfAny(s_whitespaceChars);
+            if (count < 0)
             {
-                if (!IsWhitespace(s[i]))
-                    return false;
+                return chars.Length;
             }
-            return true;
-        }
 
-        public static bool IsWhitespace(char ch)
-        {
-            return (ch <= ' ' && (ch == ' ' || ch == '\t' || ch == '\r' || ch == '\n'));
+            for (int i = count + 1; i < chars.Length; i++)
+            {
+                char c = chars[i];
```

Comment:
I'd need to look to see if the JIT will successfully eliminate the bounds check here (it might via loop cloning).  It could be worthwhile first slicing to start at count+1 so that the rest of the loop can be done in a normal 0 to Length iteration.

Reply:
Let me check, `(uint)` spam may also achieve that here